src/Raven.Server/ServerWide/SecretProtection.cs

```diff
@@ -720,6 +721,39 @@ public byte[] LoadMasterKeyFromPath()
             }
         }
 
+        internal static void ValidateExpiration(string source, ServerStore serverStore, LicenseStatus newLicenseStatus)
+        {
+            
+            var currentLicenseType = serverStore.LicenseManager.LicenseStatus.Type;
+
+            if (newLicenseStatus.Type != LicenseType.Developer) return;
+
+            var certificateNotAfter = serverStore.Server.Certificate.Certificate.NotAfter;
+            var certificateNotBefore = serverStore.Server.Certificate.Certificate.NotBefore;
+            var totalDurationInMonths = Convert.ToInt32(certificateNotAfter.Subtract(certificateNotBefore).TotalDays / 30);
```

Comment:
That is actually wrong, number of days isn't a good manner.
Use ` if (loadedCertificate.NotAfter > DateTime.UtcNow.AddMonths(4))` this logic as well.

Reply:
done